### DISCUSSION TITLE
The starter template doesn't compile

### DISCUSSION THREAD

**Discussion by pesterev** [leptos-rs/leptos#1364] [Q&A]
**Describe the bug**
I followed [this guide](https://github.com/leptos-rs/leptos#cargo-leptos) and tried to compile the starter template (`cargo leptos new --git https://github.com/leptos-rs/start`) but encountered some issues.
```rs
$ cargo leptos watch  
   Compiling quote v1.0.29
   Compiling proc-macro2 v1.0.64
   Compiling unicode-ident v1.0.10
   Compiling version_check v0.9.4
   Compiling autocfg v1.1.0
   Compiling once_cell v1.18.0
   Compiling thiserror v1.0.43
   Compiling serde v1.0.171
   Compiling syn v1.0.109
   Compiling wasm-bindgen-shared v0.2.87
   Compiling cfg-if v1.0.0
   Compiling log v0.4.19
error[E0463]: can't find crate for `core`
  |
  = note: the `wasm32-unknown-unknown` target may not be installed
  = help: consider downloading the target with `rustup target add wasm32-unknown-unknown`

error[E0463]: can't find crate for `compiler_builtins`

For more information about this error, try `rustc --explain E0463`.
error: could not compile `cfg-if` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
      Notify watching folders assets, style, src
```

**Leptos Dependencies**

The Leptos dependencies and features from my `Cargo.toml`:
```toml
leptos = { version = "0.4", features = ["nightly"] }
leptos_meta = { version = "0.4", features = ["nightly"] }
leptos_actix = { version = "0.4", optional = true }
leptos_router = { version = "0.4", features = ["nightly"] }
```

**To Reproduce**
1. rustup toolchain install nightly
2. rustup default nightly
3. rustup target add wasm32-unknown-unknown
4. cargo install cargo-leptos
5. cargo leptos new --git https://github.com/leptos-rs/start (specify any name for the project)
6. cd <project_name>
7. cargo leptos watch

**Expected behavior**
I want the starter template to be compilelable.

**Screenshots**
<img width="988" alt="image" src="https://github.com/leptos-rs/leptos/assets/8786922/26ba9cf1-810f-409e-b616-e2ae45449984">

**Additional context**
My active toolchain:
```
installed targets for active toolchain
--------------------------------------

aarch64-apple-darwin
wasm32-unknown-unknown

active toolchain
----------------

nightly-aarch64-apple-darwin
rustc 1.73.0-nightly (33a2c2487 2023-07-12)
```

1. Yes, I tried to reinstall my nightly toolchain (currently, it is `1.73.0-nightly`).
2. Yes, I tried to run `rustup target add wasm32-unknown-unknown` (it's installed).
3. I tried to write a basic project using Leptos with the `csr` feature and build using `Trunk`, but got the same errors. I feel this issue is related to my toolchain or my local configuration.
4. `cargo leptos --version`    => `cargo-leptos 0.1.11`

**Comment by luckynumberke7in** [comment]
What version of cargo-leptos are you running? There were breaking changes
in an internal rust crate that created a need to patch cargo-leptos. I
believe the working version is 0.1.11.

**Comment by gbj** [comment]
The compiler error is pretty straightforward
```
error[E0463]: can't find crate for `core`
  |
  = note: the `wasm32-unknown-unknown` target may not be installed
  = help: consider downloading the target with `rustup target add wasm32-unknown-unknown`
```
What's not straightforward is why this is happening.

I've seen people have similar issues once or twice before, in cases in which they had two separate Rust toolchains installed (for example, I think one of them had Rust installed both via Homebrew and via the Rust installer) and so they were adding the WASM target for one of them and not the other somehow. 

Not sure what the exact case is for you. I guess you can always try `cargo build --features=hydrate --target=wasm32-unknown-unknown` and see what happens. I don't think it's a `cargo-leptos` issue specifically but I may be wrong.

**Comment by pesterev** [comment]
> What version of cargo-leptos are you running? There were breaking changes in an internal rust crate that created a need to patch cargo-leptos. I believe the working version is 0.1.11.
> […](#)
> On Thu, Jul 13, 2023, 5:33 AM Vladimir Pesterev ***@***.***> wrote: *Describe the bug* I followed this guide <https://github.com/leptos-rs/leptos#cargo-leptos> and tried to compile the starter template (cargo leptos new --git https://github.com/leptos-rs/start) but encountered some issues. $ cargo leptos watch Compiling quote v1.0.29 Compiling proc-macro2 v1.0.64 Compiling unicode-ident v1.0.10 Compiling version_check v0.9.4 Compiling autocfg v1.1.0 Compiling once_cell v1.18.0 Compiling thiserror v1.0.43 Compiling serde v1.0.171 Compiling syn v1.0.109 Compiling wasm-bindgen-shared v0.2.87 Compiling cfg-if v1.0.0 Compiling log v0.4.19 error[E0463]: can't find crate for `core` | = note: the `wasm32-unknown-unknown` target may not be installed = help: consider downloading the target with `rustup target add wasm32-unknown-unknown` error[E0463]: can't find crate for `compiler_builtins` For more information about this error, try `rustc --explain E0463`.error: could not compile `cfg-if` (lib) due to 2 previous errors warning: build failed, waiting for other jobs to finish... Notify watching folders assets, style, src *Leptos Dependencies* The Leptos dependencies and features from my Cargo.toml: leptos = { version = "0.4", features = ["nightly"] }leptos_meta = { version = "0.4", features = ["nightly"] }leptos_actix = { version = "0.4", optional = true }leptos_router = { version = "0.4", features = ["nightly"] } *To Reproduce* Steps to reproduce the behavior: 1. rustup toolchain install nightly 2. rustup default nightly 3. rustup target add wasm32-unknown-unknown 4. cargo install cargo-leptos 5. cargo leptos new --git https://github.com/leptos-rs/start (specify any name for the project) 6. cd <project_name> 7. cargo leptos watch *Expected behavior* I want the starter template to be compilelable. *Screenshots* [image: image] <https://user-images.githubusercontent.com/8786922/253230920-26ba9cf1-810f-409e-b616-e2ae45449984.png> *Additional context* Add any other context about the problem here. My active toolchain: installed targets for active toolchain -------------------------------------- aarch64-apple-darwin wasm32-unknown-unknown active toolchain ---------------- nightly-aarch64-apple-darwin rustc 1.73.0-nightly (33a2c2487 2023-07-12) 1. Yes, I tried to reinstall my nightly toolchain (currently, it is 1.73.0-nightly). 2. Yes, I tried to run rustup target add wasm32-unknown-unknown (it's installed). 3. I tried to use Trunk and got the same errors. — Reply to this email directly, view it on GitHub <#1336>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AOO63SYL7NEEVGTD7KVKYQ3XP66E3ANCNFSM6AAAAAA2IVEXIQ> . You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

cargo-leptos 0.1.11

**Comment by gbj** [comment]
@pesterev What happens when you run `cargo build --features=hydrate --target=wasm32-unknown-unknown`?

**Reply by pesterev** [reply]
Got these warnings and the same errors
```bash
warning: output filename collision.
The bin target `leptos_start` in package `leptos_start v0.1.0 (~/Documents/leptos-demo)` has the same output filename as the lib target `leptos_start` in package `leptos_start v0.1.0 (~/Documents/leptos-demo)`.
Colliding filename is: ~/Documents/leptos-demo/target/wasm32-unknown-unknown/debug/leptos_start.wasm
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
warning: output filename collision.
The bin target `leptos_start` in package `leptos_start v0.1.0 (~/Documents/leptos-demo)` has the same output filename as the lib target `leptos_start` in package `leptos_start v0.1.0 (~/Documents/leptos-demo)`.
Colliding filename is: ~/Documents/leptos-demo/target/wasm32-unknown-unknown/debug/leptos_start.wasm.dwp
The targets should have unique names.
Consider changing their names to be unique or compiling them separately.
This may become a hard error in the future; see <https://github.com/rust-lang/cargo/issues/6313>.
```

**Reply by mahadeer** [reply]
I had the similar issue once even though it was working perfectly for another leptos project. Then i tried adding
`
rustup target add wasm32-unknown-unknown
`
it fixed the issue.
